File: CommandPattern(Behavioral)/CommandPatternWithHistory/undo_redo_java/Application.java
import java.util.Stack;

public class Application {
    public static void main(String[] args) {
        // Stack
        Stack<Integer> stack = new Stack<>();

        // Create History List
        HistoryList historyList = new HistoryList();

        Command pushCommand1 = new PushCommand(stack, 15);
        pushCommand1.execute();
        historyList.addCommand(pushCommand1);
        System.out.println(stack); // 15

        Command pushCommand2 = new PushCommand(stack, 16);
        pushCommand2.execute();
        historyList.addCommand(pushCommand2);
        System.out.println(stack); // 15 16

        Command popCommand = new PopCommand(stack);
        popCommand.execute();
        historyList.addCommand(popCommand);
        System.out.println(stack); // 15

        historyList.undo();
        System.out.println(stack); // 15 16

        historyList.redo(); // 15
        System.out.println(stack);

        historyList.undo(); // 15 16
        System.out.println(stack);
    }
}

File: Behavioral/CommandPattern(Behavioral)/CommandPatternWithHistory/undo_redo/Application.ts
import { HistoryList } from "./HistoryList";
import { PopCommand } from "./PopCommand";
import { PushCommand } from "./PushCommand";

export class Application {
  public static main(): void {
    const stack: number[] = []; // Create a stack
    const historyList: HistoryList = new HistoryList(); // Create a history list

    const pushc1: PushCommand = new PushCommand(stack, 6); // Create a push command
    pushc1.execute(); // Execute the push command
    historyList.addCommand(pushc1); // Add the push command to the history list
    console.log(stack); // Output: [6]

    const pushc2: PushCommand = new PushCommand(stack, 3); // Create a push command
    pushc2.execute(); // Execute the push command
    historyList.addCommand(pushc2); // Add the push command to the history list
    console.log(stack); // Output: [6, 3]

    const popc1: PopCommand = new PopCommand(stack); // Create a pop command
    popc1.execute(); // Execute the pop command
    historyList.addCommand(popc1); // Add the pop command to the history list
    console.log(stack); // Output: [6]

    historyList.undo(); // Undo the last command
    console.log(stack); // Output: [6, 3]
  }
}
